### ISSUE TITLE
Suppress warning relate defining workaround function

### ISSUE THREAD

**Issue by conao3** [purcell/package-lint#145]
Hi!
I found package-lint warn useless warning. see this [PR](https://github.com/flycheck/flycheck-cask/pull/5).
``` emacs-lisp
(eval-and-compile
  (unless (fboundp 'setq-local)
    (defmacro setq-local (var val)
      "Set variable VAR to value VAL in current buffer."
      ;; Can't use backquote here, it's too early in the bootstrap.
      (declare (debug (symbolp form)))
      (list 'set (list 'make-local-variable (list 'quote var)) val))))
```
and package-lint warn
```
flycheck…    63   1 error           "setq-local" doesn't start with package's prefix "flycheck-cask". (emacs-lisp-package)
```
This warning doesn't make sense. We check whether setq-local defined or not before define it.
Suppress this warning is also useful.

**Comment by purcell** [comment]
Nope, sorry. From the [elisp coding conventions](https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html):

> If a package needs to define an alias or a new function for compatibility with some other version of Emacs, name it with the package prefix, not with the raw name with which it occurs in the other version. Here is an example from Gnus, which provides many examples of such compatibility issues.

```el
          (defalias 'gnus-point-at-bol
            (if (fboundp 'point-at-bol)
                'point-at-bol
              'line-beginning-position))
```
